framework/DataObjects/XDataSet.py

```diff
@@ -671,10 +673,63 @@ def _changeVariableValue(self,index,var,value):
       self._data[var].values[index] = value
     # if it's in the collector ...
     elif index < lenColl + lenData:
-      self._collector[index][self._allvars.index(var)] = value
+      self._collector[index][self._orderedVars.index(var)] = value
     else:
       self.raiseAnError(IndexError,'Requested value change for realization "{}", which is past the end of the data object!'.format(index))
 
+  def _checkAlignedIndexes(self,rlz,tol=1e-15):
+    """
+      Checks to see if indexes should be stored as "aligned" or if they need to be stored distinctly.
+      @ In, rlz, dict, formatted realization with either singular or np arrays as values
+      @ In, tol, float, optional, matching tolerance
+      @ Out, None
+    """
+    for index in self.indexes:
+      # if it's aligned so far, check if it still is
+      if index in self._alignedIndexes.keys():
+        # if close enough, then keep the aligned values; otherwise, take action
+        if isinstance(rlz[index][0],(float,int)):
+          closeEnough = all(np.isclose(rlz[index],self._alignedIndexes[index],rtol=tol))
+        else:
+          closeEnough = all(rlz[index] == self._alignedIndexes[index])
+        if not closeEnough:
+          dtype = rlz[index].dtype
+          # TODO add new column to collector, propagate values up to (not incl) current rlz
+          self.raiseAWarning('A mismatch in time scales has been found between consecutive realizations.  Consider synchronizing before doing any postprocessing!')
```

Comment:
this warning is going to be printed everytime we get a realization from a code (almost).
Do we need it herE?

Reply:
Nope, we don't need to.  We could either create a "mismatch noticed" flag that prints ... umm ... some other time, end of step on a call maybe?  Or we could avoid printing it entirely.

My thought was only to save the user some pain in not realizing synchronization was needed.  I can remove this if you like.